### PR TITLE
feat: add support for output.crossOriginLoading config

### PR DESCRIPTION
**Pull request by chenjiahan**
## Summary

Add support for output.crossOriginLoading config, same as webpack：[crossOriginLoading](https://webpack.js.org/configuration/output/#outputcrossoriginloading).

Document: https://github.com/web-infra-dev/rspack-website/pull/255

### anonymous

```js
module.exports = {
  output: {
    crossOriginLoading: "anonymous"
  }
};
```

### use-credentials

```js
module.exports = {
  output: {
    crossOriginLoading: "use-credentials"
  }
};
```

## Related issue (if exists)

- Closes https://github.com/web-infra-dev/rspack/issues/2478

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [x] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [x] I have added changeset via `pnpm run changeset`.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: crates/rspack_plugin_runtime/src/runtime_module/load_script.rs [hunk 1]
```diff
@@ -24,8 +24,12 @@ impl RuntimeModule for LoadScriptRuntimeModule {
     self.id
   }
 
-  fn generate(&self, _compilation: &Compilation) -> BoxSource {
-    RawSource::from(include_str!("runtime/load_script.js")).boxed()
+  fn generate(&self, compilation: &Compilation) -> BoxSource {
+    RawSource::from(include_str!("runtime/load_script.js").replace(
+      "__CROSS_ORIGIN_LOADING_PLACEHOLDER__",
+      &compilation.options.output.cross_origin_loading.to_string(),
```
Comment (underfin) [inline crates/rspack_plugin_runtime/src/runtime_module/load_script.rs:30]:
```suggestion
      &compilation.options.output.cross_origin_loading,
```

File: crates/rspack_plugin_runtime/src/runtime_module/css_loading.rs [hunk 2]
```diff
@@ -101,7 +101,12 @@ impl RuntimeModule for CssLoadingRuntimeModule {
         &stringify_chunks(&initial_chunk_ids_with_css, 0)
       )));
 
-      source.add(RawSource::from(include_str!("runtime/css_loading.js")));
+      source.add(RawSource::from(
+        include_str!("runtime/css_loading.js").replace(
+          "__CROSS_ORIGIN_LOADING_PLACEHOLDER__",
+          &compilation.options.output.cross_origin_loading.to_string(),
```
Comment (underfin) [inline crates/rspack_plugin_runtime/src/runtime_module/css_loading.rs:107]:
```suggestion
          &compilation.options.output.cross_origin_loading,
```

**Review by underfin — DISMISSED** [review]

**Comment by underfin** [comment]
Webpack has `test/configCases/crossorigin/set-crossorigin` test, if add it will be better.

File: crates/rspack_plugin_runtime/src/runtime_module/load_script.rs [hunk 3]
```diff
@@ -24,8 +24,12 @@ impl RuntimeModule for LoadScriptRuntimeModule {
     self.id
   }
 
-  fn generate(&self, _compilation: &Compilation) -> BoxSource {
-    RawSource::from(include_str!("runtime/load_script.js")).boxed()
+  fn generate(&self, compilation: &Compilation) -> BoxSource {
+    RawSource::from(include_str!("runtime/load_script.js").replace(
+      "__CROSS_ORIGIN_LOADING_PLACEHOLDER__",
+      &compilation.options.output.cross_origin_loading,
```
Comment (underfin) [inline crates/rspack_plugin_runtime/src/runtime_module/load_script.rs:30]:
```suggestion
      &compilation.options.output.cross_origin_loading.to_string(),
```

File: crates/rspack_plugin_runtime/src/runtime_module/css_loading.rs [hunk 4]
```diff
@@ -101,7 +101,12 @@ impl RuntimeModule for CssLoadingRuntimeModule {
         &stringify_chunks(&initial_chunk_ids_with_css, 0)
       )));
 
-      source.add(RawSource::from(include_str!("runtime/css_loading.js")));
+      source.add(RawSource::from(
+        include_str!("runtime/css_loading.js").replace(
+          "__CROSS_ORIGIN_LOADING_PLACEHOLDER__",
+          &compilation.options.output.cross_origin_loading,
```
Comment (underfin) [inline crates/rspack_plugin_runtime/src/runtime_module/css_loading.rs:107]:
```suggestion
          &compilation.options.output.cross_origin_loading.to_string(),
```

**Review by underfin — APPROVED** [review]

**Comment by chenjiahan** [comment]
> Webpack has test/configCases/crossorigin/set-crossorigin test, if add it will be better.

I have tried to enable that case, but it seems like some runtime methods is not implemented yet, such as `__non_webpack_require__`. 

Maybe we can enable it after `__non_webpack_require__` is added.

image

File: crates/rspack_binding_options/src/options/raw_output.rs [hunk 5]
```diff
@@ -71,6 +71,30 @@ impl From<RawLibraryOptions> for LibraryOptions {
   }
 }
 
+#[derive(Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+#[napi(object)]
+pub struct RawCrossOriginLoading {
+  #[napi(ts_type = "boolean | string")]
```
Comment (h-a-n-a) [inline crates/rspack_binding_options/src/options/raw_output.rs:78]: The typing here seems differentiated from the usage `"bool" | "string"`

**Pull request merged**